README.md

```diff
@@ -98,42 +88,341 @@ See [variables.tf] and [examples/] for details and use-cases.
+  Default is `{}`.
+
+##### Internet Gateway
+
+- **`internet_gateway_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created Internet Gateway.
+  An Internet Gateway is created if a public subnet is defined.
+  All public egress and ingress traffic of all public subnets will be routed through the same Internet Gateway.
+  Default is `{}`.
+
+##### NAT Gateways
+
+- **`nat_gateway_mode`**: *(Optional `string`)*
+
+  Set the mode for the NAT Gateways. NAT Gateways will only be created when private subnets are defined.
+  Each private subnet needs at least one configured public subnet in the same availability zone.
+  Each NAT gateway will be assigned an Elastic IP Address.
+  When changing the mode, NAT Gateways and Elastic IP Addresses will be created or destroyed.
+  All public egress traffic of all private subnets will be routed through the same set of NAT Gateways.
+  Possible modes are `none`, `single`, or `one_per_az`. Choose
+
+  - `none` to create no NAT Gateways at all (use for debugging only),
+  - `single` to create a single NAT Gateway inside the first defined Public Subnet, or
+  - `one_per_az` to create one NAT Gateway inside the first Public Subnet in each Availability Zone.
+
+  Default is `"single"`.
+
+- **`nat_gateway_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created NAT Gateways.
+  Default is `{}`.
+
+- **`eip_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created NAT Gateway Elastic IP Addresses.
+  Default is `{}`.
+
+##### Subnet Routing
+
+- **`route_table_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to all created Route Tables.
+  Default is `{}`.
+
+- **`public_route_table_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created Public Route Tables.
+  Default is `{}`.
+
+- **`private_route_table_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created Private Route Tables.
+  Default is `{}`.
+
+- **`intra_route_table_tags`**: *(Optional `map(string)`)*
+
+  A map of tags to apply to the created Intra Route Tables.
+  Default is `{}`.
 
 ## Module Attributes Reference
 
 The following attributes are exported by the module:
 
-- **`module_enabled`**
+- **`public_subnets_by_group`**
 
-  Whether this module is enabled.
+  A map of lists of public subnets keyed by group. (aws_subnet)
 
-- **`module_defaults`**
+- **`public_route_table_ids_by_group`**
 
-  Default settings that overwrite the module and resource defaults in this module.
+  A map of lists of public route table IDs keyed by group.
 
-- **`module_tags`**
+- **`public_subnet_ids_by_group`**
 
-  The map of tags that are being applied to all created resources that accept tags.
+  A map of lists of public subnet IDs keyed by group.
+
+- **`private_subnets_by_group`**
+
+  A map of lists of private subnets keyed by group. (aws_subnet)
```

Comment:
maybe link aws_subnet to the right terraform doc?

Reply:
way too much effort without a big win atm..  lmgtf.